cmd/agent/main.go

```diff
@@ -21,37 +21,56 @@
 package main
 
 import (
-	"flag"
 	"runtime"
 
 	"github.com/uber/jaeger-lib/metrics/go-kit"
 	"github.com/uber/jaeger-lib/metrics/go-kit/expvar"
 	"go.uber.org/zap"
 
 	"github.com/uber/jaeger/cmd/agent/app"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
-func main() {
-	builder := app.NewBuilder()
-	builder.Bind(flag.CommandLine)
-	flag.Parse()
+var (
+	builder = app.NewBuilder()
+)
+
+func init() {
```

Comment:
var and init() change from using local vars to using global vars, seems like a step backwards. It doesn't look like you need to do that, everything can still be local to main()